pkg/certificate/certificate.go

```diff
@@ -116,40 +194,110 @@ func (k *Keypair) generate() error {
 		return err
 	}
 
-	keyFile, err := os.Create(k.keyFile)
+	keyFile, err := os.OpenFile(kf, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
```

Comment:
Suppose you could ignore my dir permissions comment above, since the key itself is not readable by the group.